pointmatcher/IO.cpp

```diff
@@ -1573,21 +1574,94 @@ typename PointMatcherIO<T>::DataPoints PointMatcherIO<T>::loadPLY(std::istream&
+                    value = static_cast<T>(temp);
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::INT16:
+                {
+                    int16_t temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(int16_t));
+                    value = static_cast<T>(temp);
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::UINT16:
+                {
+                    uint16_t temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(uint16_t));
+                    value = static_cast<T>(temp);
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::INT32:
+                {
+                    // TODO what happens if T is float and we overflow?
+                    int32_t temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(int32_t));
+                    value = static_cast<T>(temp);
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::UINT32:
+                {
+                    // TODO what happens if T is float and we overflow?
+                    uint32_t temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(uint32_t));
+                    value = static_cast<T>(temp);
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::FLOAT32:
+                {
+                    float temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(float));
+                    value = static_cast<T>(temp); // Directly assign if T is float, or cast if T is double
+                    break;
+                }
+                case PLYProperty::PLYPropertyType::FLOAT64:
+                {
+                    // TODO what happens if T is float and we read a double?
+                    double temp;
+                    is.read(reinterpret_cast<char*>(&temp), sizeof(double));
+                    value = static_cast<T>(temp); // Directly assign if T is double, or cast if T is float
+                    break;
+                }
+                default:
+                {
+                    throw runtime_error(
+                        (boost::format("Unsupported data type in binary mode %1%.") % static_cast<int>(vertex->properties[propID].type)).str());
+                }
+            }
+        }
+        if ((is_binary && !is) || (!is_binary && !(is >> value)))
+        {
+            throw runtime_error(
+                    (boost::format("PLY parse error: expected %1% values (%2% points with %3% properties) but only found %4% values.") % nbValues % nbPoints % nbProp %
+                     i).str());
+        }
+        else
+        {
+            const int row = vertex->properties[propID].pmRowID;
+            const PMPropTypes type = vertex->properties[propID].pmType;
+
+            // rescale color from [0,254] to [0, 1[
```

Comment:
The highest value in an 8-bit unsigned integer is 255, why does it end at 254?

Reply:
I think it's just a typo since the actual scaling uses 255.